library/kani_macros/src/sysroot/contracts.rs

```diff
@@ -218,7 +337,7 @@ impl VisitMut for PostconditionInjector {
 /// - Creates new names for them;
 /// - Replaces all occurrences of those idents in `attrs` with the new names and;
 /// - Returns the mapping of old names to new names.
-fn rename_argument_occurences(sig: &syn::Signature, attr: &mut Expr) -> HashMap<Ident, Ident> {
+fn rename_argument_occurrences(sig: &syn::Signature, attr: &mut Expr) -> HashMap<Ident, Ident> {
```

Comment:
Can you please create an issue to rename arguments (or trigger an error) if there's a name conflict with `result` and any other special variable names required by contracts?

Reply:
Already exists 😛  https://github.com/model-checking/kani/issues/2597

Reply:
Also this renaming is safe, because we use `Span::mixed_site` which is subject to hygiene rules.

Reply:
Oh, what does `mixed_site` do? I don't know this one.

Reply:
Yeah it's a bad naming. If you use `mixed_site` on an ident for a variable, then that variable is only visible to other code generated by this macro. If you use `call_site` it is also visible to user-written code. 

The reason it's called "mixed site" is that those rules only apply to variables (and maybe a few more things) but not to everything, so i think if you define new types for instance, they are always visible like with `call_site`.

Reply:
Yeah so according to https://docs.rs/proc-macro2/1.0.67/proc_macro2/struct.Span.html#method.mixed_site it basically uses `macro_rules` rules, where variable identifiers are specific to the macro invocation site and only visible there.